bug: not handle column non-existent in mssql

### What happened?

When I use  function `sql()` of `mssql` with a non-exist column like below
``` python
ibis.mssql.connect(...).sql('SELECT x from tpch.tiny.orders')
```
ibis will throw `'NoneType' object has no attribute 'lower'`
Because the function will get schema by `_get_schema_using_query()`

https://github.com/ibis-project/ibis/blob/ea1c179036b10b53e58156a0bfc3e31f3472a388/ibis/backends/mssql/__init__.py#L307-L335

But the rows will get `[(None, None, None, None, None), (None, None, None, None, None)]` and  put the `None` into `self.compiler.type_mapper.from_string()`

Maybe we should check if the value is None or not.

### What version of ibis are you using?

9.5.0

### What backend(s) are you using, if any?

mssql

### Relevant log output

```sh
../.venv/lib/python3.11/site-packages/ibis/backends/sql/__init__.py:176: in sql
    schema = self._get_schema_using_query(query)
../.venv/lib/python3.11/site-packages/ibis/backends/mssql/__init__.py:333: in _get_schema_using_query
    newtyp = self.compiler.type_mapper.from_string(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'ibis.backends.sql.datatypes.MSSQLType'>, text = None
nullable = None

    @classmethod
    def from_string(cls, text: str, nullable: bool | None = None) -> dt.DataType:
>       if dtype := cls.unknown_type_strings.get(text.lower()):
E       AttributeError: 'NoneType' object has no attribute 'lower'

../.venv/lib/python3.11/site-packages/ibis/backends/sql/datatypes.py:195: AttributeError
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/ibis/backends/mssql/__init__.py
+++ b/ibis/backends/mssql/__init__.py
@@ -321,7 +321,9 @@ class Backend(SQLBackend, CanCreateCatalog, CanCreateDatabase):
           is_nullable,
           system_type_name,
           precision,
-          scale
+          scale,
+          error_number,
+          error_message
         FROM sys.dm_exec_describe_first_result_set({tsql}, NULL, 0)
         ORDER BY column_ordinal
         """
@@ -329,7 +331,18 @@ class Backend(SQLBackend, CanCreateCatalog, CanCreateDatabase):
             rows = cur.fetchall()
 
         schema = {}
-        for name, nullable, system_type_name, precision, scale in rows:
+        for (
+            name,
+            nullable,
+            system_type_name,
+            precision,
+            scale,
+            err_num,
+            err_msg,
+        ) in rows:
+            if err_num is not None:
+                raise com.IbisInputError(f".sql failed with message: {err_msg}")
+
             newtyp = self.compiler.type_mapper.from_string(
                 system_type_name, nullable=nullable
             )

--- a/ibis/backends/mssql/tests/test_client.py
+++ b/ibis/backends/mssql/tests/test_client.py
@@ -8,6 +8,7 @@ import sqlglot.expressions as sge
 from pytest import param
 
 import ibis
+import ibis.common.exceptions as com
 import ibis.expr.datatypes as dt
 from ibis import udf
 from ibis.backends.mssql.tests.conftest import (
@@ -261,6 +262,11 @@ def test_dot_sql_with_unnamed_columns(con):
 
     df = expr.execute()
     assert len(df) == 1
+
+
+def test_dot_sql_error_handling(con):
+    with pytest.raises(com.IbisInputError, match="Invalid column name"):
+        con.sql("SELECT not_a_column")
 
 
 @pytest.mark.parametrize(

